docs/reference/modules/cluster/remote-clusters-security.asciidoc

```diff
@@ -0,0 +1,41 @@
+[[remote-clusters-security]]
+=== Configure remote clusters with security
+To use {ccr} or {ccs} with secured clusters, all clusters must have the {es} 
+security features enabled.
+
+The local cluster must be allowed to connect to the remote clusters, which means 
+that the certificate authority (CA) used to sign the keystore of the local
+cluster must be trusted by the remote clusters.
+
+User authentication is performed on the local cluster and the user and user’s 
+roles are passed to the remote clusters. A remote cluster checks the user’s
+roles against its local role definitions to determine which indices the user is 
+allowed to access.
+
+Before using {ccr} or {ccs} with secure {es} clusters, complete the following 
+configuration tasks:
+
+. Enable the {es} {security-features} on every node in each connected cluster by
+setting `xpack.security.enabled` to `true`. Refer to the
+<<general-security-settings,{es} security settings>>.
+
+. Encrypt internode communications with Transport Layer Security (TLS) on every
+node to establish a trust relationship between the local cluster and all remote
+clusters. Refer to 
+<<security-basic-setup,set up basic security for the {stack}>> for the required
+steps to configure security.
++
+NOTE: This procedure uses the same CA to generate certificates for all nodes.
+Alternatively, you can add the CA certificates from the local cluster as a
+trusted CA in each remote cluster. 
```

Comment:
Hmm...interesting. Connections to a remote cluster are typically unidirectional, correct? Meaning you connect to remote clusters from a local cluster. But if a remote cluster trusts its CA certificates, and those are trusted by the local cluster, then all connected clusters (both local and remote) are trusting the same certificates. We should test both scenarios so that users are aware.

Reply:
> Connections to a remote cluster are typically unidirectional, correct?

It appears to be unidirectional because the conceptual data flow is one way (especially for CCR). But cross cluster communication uses the same Elasticsearch Transprot layer protocol. This means if the transport layer is configured to require mutual TLS authentication (and it should based on our recommended best practice), remote and local nodes are treated no differently at the transport layer. When establishing conneciton, each  party will verify certificate from the other side. Thus they require trust each other's certificate or CA.

You can essentially picture those nodes (local and remote) forming a single cluster at transport layer. Indeed, CCS effectively works that way, i.e. it treats shards on the remote nodes the same way as local shards.

In summary, setting up remote cluster really means full mutual trust. They are essentially the same security domain. There were questions around whether we can separate them at both the TLS and RABC layers. I think we want to make improvements. But there has not been any formal planning yet.